app/models/grda_warehouse/vispdat.rb

```diff
@@ -0,0 +1,252 @@
+module GrdaWarehouse
+  class Vispdat < GrdaWarehouseBase
+    ####################
+    # Constants
+    ####################
+    US_PHONE_NUMBERS = /\A(\+1)?\(?(\d{3})\)?\s*-?\s*(\d{3})\s*-?\s*(\d{4})\s*-?\s*\z/
+
+    ####################
+    # enums
+    ####################
+    enum language_answer: [:language_english, :language_spanish, :language_french]
+    enum sleep_answer: [:sleep_shelters, :sleep_transitional_housing, :sleep_safe_haven, :sleep_outdoors, :sleep_other, :sleep_refused]
+
+    %w(attacked threatened legal tricked risky owe_money get_money activities basic_needs abusive leave chronic hiv disability avoid_help pregnant eviction drinking mental head learning brain medication sell trauma picture).each do |field|
+      enum "#{field}_answer".to_sym => { "#{field}_answer_yes".to_sym => 1, "#{field}_answer_no".to_sym => 0, "#{field}_answer_refused".to_sym => 2 }
+    end
+
+    enum when_answer: [:morning, :afternoon, :evening, :night]
+
+    ####################
+    # Associations
+    ####################
+    belongs_to :client, class_name: 'GrdaWarehouse::Hud::Client'
+
+    ####################
+    # Behaviors
+    ####################
+    has_paper_trail
+
+    ####################
+    # Validations
+    ####################
+    validates :first_name, presence: true, length: { in: 2..30 }
```

Comment:
The various names, SSN and DOB should all be obtained directly from the related client and not stored or validated here.